lib/main.js

```diff
@@ -190,6 +190,47 @@ proto.playback = {
     this.playback._sliderEl.dispatchEvent(evt);
   },
 
+  // Return song duration in milliseconds
+  getSongDuration: function () {
+    return this.playback._sliderEl.max;
+  },
+
+  // PlayPause element states:
+  //   PLAYING: {__data__: {icon: 'av:pause-circle-filled'}, disabled: false}
+  //   PAUSED: {__data__: {icon: 'av:sj:pause-circle-fill'}, disabled: false}
+  //   STOPPED: {__data__: {icon: 'av:sj:play-circle-fill'}, disabled: true}
+  getPlaybackState: function () {
+    if (!this.playback._playPauseEl.disabled) {
+      if (this.playback._playPauseEl.__data__.icon === 'av:pause-circle-filled') {
+        return GMusic.Playback.PLAYING;
+      } else {
+        return GMusic.Playback.PAUSED;
+      }
+    } else {
+      return GMusic.Playback.STOPPED;
+    }
+  },
+
+  getSongInfo: function () {
+    var songInfo = {};
+    songInfo.title = this.doc.getElementById(SELECTORS.info.titleId).textContent;
+    songInfo.artist = this.doc.getElementById(SELECTORS.info.artistId).textContent;
+    songInfo.album = this.doc.querySelector(SELECTORS.info.albumSelector).textContent;
+    songInfo.art = this.doc.getElementById(SELECTORS.info.albumArtId);
+    songInfo.duration = this.doc.getElementById(SELECTORS.playback.sliderId).max;
+
+    songInfo.title = (songInfo.title) ? songInfo.title : 'Unknown';
+    songInfo.artist = (songInfo.artist) ? songInfo.artist : 'Unknown';
+    songInfo.album = (songInfo.album) ? songInfo.album : 'Unknown';
+    songInfo.art = (songInfo.art) ? songInfo.art.src : null;
+
+    // The art may be a protocol-relative URL, so normalize it to HTTPS
+    if (songInfo.art && songInfo.art.slice(0, 2) === '//') {
+      songInfo.art = 'https:' + songInfo.art;
+    }
+    return songInfo;
+  },
```

Comment:
We grabbed this logic from the `change:song` logic right? Can we call this function in that listener?